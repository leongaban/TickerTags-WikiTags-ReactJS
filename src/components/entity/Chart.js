import React from 'react'
import { type } from 'ramda'
import { getChartData, getCloudData } from '../../data/api'

const Login = () => <div className="well"><p className="text-center mt10">To view social mentions please <a href="/wiki/login">Log In</a></p></div>
const socialColors = {
    twitter: '#0084b4',
    boards: '#3b5998',
    video_comment: '#bb0000',
    blogs: '#f26522',
    reviews: '#ffd400',
    news: '#6a737b'
}
const inactiveColor = '#eeeeee';
const setSocialColor = (btn, state) => {
    if (btn == state) return socialColors[state];
    return inactiveColor;
};

class Social extends React.Component {

    constructor(props) {
        super(props)
        this.onSelection = this.onSelection.bind(this);
    }

    onSelection(media) {
        
    }

    render() {
        const { media, text, color, icon, onSelect } = this.props;
        const iconClass = `fa ${icon} fa-stack-1x fa-inverse`;
        const iconStyle = { color };
        const socialStyle = { cursor: 'pointer', decoration: 'none' };

        return (
            <a className="mr10" onClick={() => onSelect(media)} style={socialStyle}>
                <span className="fa-stack fa-lg mr5">
                    <i id={media} className="fa fa-circle fa-stack-2x" style={iconStyle}/>
                    <i className={iconClass}/>
                </span>
                { text }
            </a>
        )
    }
}

export default class Chart extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            media: 'twitter',
            days: 30,
            id: `chart-${this.props.subject.id}`,
            authenticated: false
        }

        this.getSocialData = this.getSocialData.bind(this);
        this.getCloudData = getCloudData.bind(this);
    }

    // getSocialData({media, days}) {
    getSocialData(state) {
        // this.getCloudData(days)
        // this.getCloudData().then(console.log)
        const element = document.getElementById(this.state.id);
        getChartData(this.props.subject.id, element, state.days);

        if (type(state) === 'String') {
            this.setState({
                media: state
            });
        }
    }

    componentDidMount() {
        // Load the Visualization API and the corechart package.
        // console.log('Chart mounted');
        google.charts.load('current', { packages: ['corechart', 'line'] });
        // google.charts.setOnLoadCallback(this.getSocialData({ days: this.state.days }));
        google.charts.setOnLoadCallback(this.getSocialData(this.state));
    }

    render() {
        const chartStyle = {
            width: '100%',
            height: '300px',
            backgroundImage: 'url(http://distribution.wikitags.com/imgs/wikitags-watermark-10.png)',
            backgroundRepeat: 'no-repeat',
            backgroundPosition: 'center',
            backgroundSize: '40%'
        };

        const { subject } = this.props;
        const media = this.state.media;

        return (
            <div>
                <div id={ this.state.id } className="chart_div" style={ chartStyle }/>
                <div className="row socialmentions" id={ subject.id }>
                    <div className="mb15">
                        <Social onSelect={ this.getSocialData }
                                color={ setSocialColor('twitter', media) }
                                media="twitter" text="twitter" icon="fa-twitter"/>

                        <Social onSelect={ this.getSocialData }
                                color={ setSocialColor('board', media) }
                                media="board" text="boards" icon="fa-comments"/>

                        <Social onSelect={ this.getSocialData }
                                color={ setSocialColor('video_comment', media) }
                                media="video_comment" text="video comments" icon="fa-youtube-play"/>

                        <Social onSelect={ this.getSocialData }
                                color={ setSocialColor('blog', media) }
                                media="blog" text="blogs" icon="fa-rss"/>

                        <Social onSelect={ this.getSocialData }
                                color={ setSocialColor('review', media) }
                                media="review" text="reviews" icon="fa-star"/>

                        <Social onSelect={ this.getSocialData }
                                color={ setSocialColor('news', media) }
                                media="news" text="news" icon="fa-newspaper-o"/>
                    </div>
                    { this.state.authenticated ? <hr id="social_mentions"/> : <Login/> }
                </div>
            </div>
        )
    }
}