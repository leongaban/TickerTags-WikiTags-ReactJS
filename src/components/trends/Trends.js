import React from 'react'
import Header from '../common/Header'
import Footer from '../common/Footer'
import Chart from '../entity/Chart'

const handleSubmit = () => {

};

const getChartTrendsData = (range) => {
	// console.log('getChartTrendsData range', range);
};

export default function Trends({ body }) {
    // console.log('inside the Trends component, body:', body);
    const subject = body.subject ? body.subject : { id: 0 };
    // console.log('subject', subject);

    return (
        <div className="trends-component">
        	<div className="container">
	            <div className="row">
		        	<Header />
		        	<h2>Trends <small>compare social volumes of up to three terms</small></h2>
		        	<div className="form-inline">
		        		<form onSubmit={ () => handleSubmit() }>
							<label className="sr-only" htmlFor="inlineFormInput">Term 1</label>
							<input id="trends_term_1" type="text" className="form-control mb-2 mr-sm-2 mb-sm-0" id="inlineFormInput" placeholder="Term 1"/>
							<label className="sr-only" htmlFor="inlineFormInput">Term 2</label>
							<input id="trends_term_2" type="text" className="form-control mb-2 mr-sm-2 mb-sm-0" id="inlineFormInput" placeholder="Term 2"/>
							<label className="sr-only" htmlFor="inlineFormInput">Term 3</label>
							<input id="trends_term_3" type="text" className="form-control mb-2 mr-sm-2 mb-sm-0" id="inlineFormInput" placeholder="Term 3"/>
							<button type="submit" className="btn btn-primary" onClick={ () => getChartTrendsData() }>Submit</button>
						</form>
		        	</div>
		        	<div className="btn-group" role="group">
						<button type="button" id="chart_select_90" className="btn btn-default"
								onClick={ () => getChartTrendsData(90) }>90 Days</button>
						<button type="button" id="chart_select_60" className="btn btn-default"
								onClick={ () => getChartTrendsData(60) }>60 Days</button>
						<button type="button" id="chart_select_30" className="btn btn-default active"
								onClick={ () => getChartTrendsData(30) }>30 Days</button>
					</div>
				    <Chart subject={ subject }/>
			    </div>
			</div>
	    </div>
    );
}