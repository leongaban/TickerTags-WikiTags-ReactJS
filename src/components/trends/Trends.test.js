import React from 'react'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Trends from './Trends'
import Chart from '../entity/Chart'

const body = { subject: { id: 0 } };
const TrendComponent = shallow(<Trends body={body}/>);

describe('<Trends />', () => {
    it('renders', () => {
        const tree = toJson(TrendComponent);
        expect(tree).toMatchSnapshot(TrendComponent);
    });

    it('contains the Chart component', () => {
        expect(TrendComponent.find(Chart).length).toBe(1);
    });
});