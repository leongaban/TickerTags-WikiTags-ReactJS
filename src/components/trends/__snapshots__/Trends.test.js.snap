// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  className="trends-component"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <Header />
      <h2>
        Trends 
        <small>
          compare social volumes of up to three terms
        </small>
      </h2>
      <div
        className="form-inline"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            className="sr-only"
            htmlFor="inlineFormInput"
          >
            Term 1
          </label>
          <input
            className="form-control mb-2 mr-sm-2 mb-sm-0"
            id="inlineFormInput"
            placeholder="Term 1"
            type="text"
          />
          <label
            className="sr-only"
            htmlFor="inlineFormInput"
          >
            Term 2
          </label>
          <input
            className="form-control mb-2 mr-sm-2 mb-sm-0"
            id="inlineFormInput"
            placeholder="Term 2"
            type="text"
          />
          <label
            className="sr-only"
            htmlFor="inlineFormInput"
          >
            Term 3
          </label>
          <input
            className="form-control mb-2 mr-sm-2 mb-sm-0"
            id="inlineFormInput"
            placeholder="Term 3"
            type="text"
          />
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-default"
          id="chart_select_90"
          onClick={[Function]}
          type="button"
        >
          90 Days
        </button>
        <button
          className="btn btn-default"
          id="chart_select_60"
          onClick={[Function]}
          type="button"
        >
          60 Days
        </button>
        <button
          className="btn btn-default active"
          id="chart_select_30"
          onClick={[Function]}
          type="button"
        >
          30 Days
        </button>
      </div>
      <Chart
        subject={
          Object {
            "id": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
